app/js/listviewrowclick.js

```diff
@@ -0,0 +1,75 @@
+var BfsListViewRowClick = BfsListViewRowClick || {};
+
+BfsListViewRowClick.addClassicMutationObserver = function () {
+  var container = document;
+  var observer = new MutationObserver(BfsListViewRowClick.addRowOnClickCheckboxEventHandlerWhenListViewRowsAdded);
+
+  observer.observe(container, {
+    childList: true,
+    attributes: false,
+    characterData: false,
+    subtree: true,
+    attributeOldValue: false,
+    characterDataOldValue: false
+  });
+}
+
+BfsListViewRowClick.addRowOnClickCheckboxEventHandlerWhenListViewRowsAdded = function (mutations) {
+  // Since this code executes on each DOM addition or removal to the document,
+  // this low-level code was added to be as performant as possible.
+  var mutationsLength = mutations.length;
+  for (var mutationIndex = 0; mutationIndex < mutationsLength; ++mutationIndex) {
```

Comment:
Also, because you're doing ++mutationIndex, do we ever hit the first index of 0 here?